mkat_fpga_tests/test_cbf.py

```diff
@@ -748,37 +748,55 @@ def test_sensor_values(self):
                 'Sensor status fail: {}, {} '
                     .format(sensor.name, sensor.get_status()))
 
-    @aqf_vr('TP.C.dummy_vr_5')
+    @aqf_vr('TP.C.1.16')
     def test_roach_qdr_sensors(self):
-        """ """
-        an_e = threading.Event()
-        def event_(an_e, *args):
-            print 'Event occured'
-            try:
-                an_e.set()
-            except Exception, exc:
-                print exc
-        an_event = partial(event_, an_e)
-
+        """Sensor QDR memory error"""
         array_sensors = correlator_fixture.katcp_rct.sensor
+        # Select a host
         xhost = self.correlator.xhosts[0]
+        Aqf.step("Selected host: {}".format(xhost.host))
+        host_sensor =  getattr(array_sensors, '{}_xeng_qdr'.format(
+                            xhost.host.lower()))
+        # Check if qdr is okay
+        Aqf.is_true(host_sensor.get_value(), 'Check the QDR status: {} on {}.'
+            .format(host_sensor.get_status(),xhost.host))
+        sensor_timeout = 10
+        host_sensor.set_strategy('auto')
+        Aqf.step("Writing junk to {} memory.".format(xhost.host))
+        # Write junk to memory
         xhost.blindwrite('qdr1_memory', 'write_junk_to_memory')
-        Aqf.is_true(
-            array_sensors.roach020a0a_xeng_qdr.get_value() == xhost.qdr_okay(),
-                'Check that the memory is corrupted.')
-
-        Aqf.is_true(array_sensors.roach020a0a_xeng_qdr.get_value(),
-            'Check that the memory recovered successfully.')
-        array_sensors.roach020a0a_xeng_qdr.set_strategy('auto')
-        array_sensors.roach020a0a_xeng_qdr.register_listener(an_event)
-
-        Aqf.is_true(array_sensors.roach020a0a_xeng_qdr.get_value(),
-            'Check that the memory recovered successfully.')
-
-        xhost.vacc_get_error_detail()[1]['parity']
-
-        self.addCleanup(array_sensors.roach020a0a_xeng_qdr.unregister_listener(an_event))
-        import IPython;IPython.embed()
+        try:
+            if host_sensor.wait(False, timeout=10):
+                # Verify that qdr corrupted or unreadable
+                Aqf.equals(host_sensor.get_status(), 'error',
+                    'Confirm that the memory on {} is unreadable/corrupted.'
+                        .format(xhost.host))
+            else:
+                Aqf.failed('The memory is not unreadable/corrupted.')
```

Comment:
Perhaps change your strings to say: "Confirm that sensor indicates that memory is unreadable..." and similar for the failed case.  Also, perhaps only catch the timeout error, since that is going to be the common case?